src/EditorFeatures/CSharp/RawStringLiteral/RawStringLiteralCommandHandler_Return.cs

```diff
@@ -50,12 +51,164 @@ public bool ExecuteCommand(ReturnKeyCommandArgs args, CommandExecutionContext co
+        if (currentSnapshot[position] == '"')
+        {
+            return HandleCaretOnQuote(textView, subjectBuffer, span, position, context.OperationContext.UserCancellationToken);
+        }
+
+        return HandleCaretNotOnQuote(textView, subjectBuffer, span, position, context.OperationContext.UserCancellationToken);
+    }
+
+    private bool HandleCaretNotOnQuote(ITextView textView, ITextBuffer subjectBuffer, SnapshotSpan span, int position, CancellationToken cancellationToken)
+    {
+        // If the caret is not on a quote, we need to find whether we are within the contents of a single-line raw string literal
+        // but not inside an interpolation
+        // If we are inside a raw string literal and the caret is not on top of a quote, it is part of the literal's text
+        // Here we try to ensure that the literal's closing quotes are properly placed in their own line
+        // We could reach this point after pressing enter on a single-line raw string
+
+        var currentSnapshot = subjectBuffer.CurrentSnapshot;
+        var document = currentSnapshot.GetOpenDocumentInCurrentContextWithChanges();
+        if (document == null)
             return false;
 
+        var parsedDocument = ParsedDocument.CreateSynchronously(document, cancellationToken);
+
+        var token = parsedDocument.Root.FindToken(position);
+        switch (token.Kind())
+        {
+            case SyntaxKind.SingleLineRawStringLiteralToken:
+                break;
+
+            case SyntaxKind.InterpolatedStringTextToken
+            when token.Parent?.Parent is InterpolatedStringExpressionSyntax interpolated &&
+                interpolated.StringStartToken.Kind() is SyntaxKind.InterpolatedSingleLineRawStringStartToken:
+                break;
+
+            default:
+                return false;
+        }
+
+        var indentationOptions = subjectBuffer.GetIndentationOptions(_editorOptionsService, document.Project.GetFallbackAnalyzerOptions(), document.Project.Services, explicitFormat: false);
+        var indentation = token.GetPreferredIndentation(parsedDocument, indentationOptions, cancellationToken);
+
+        var newLine = indentationOptions.FormattingOptions.NewLine;
+
+        using var transaction = CaretPreservingEditTransaction.TryCreate(
+            CSharpEditorResources.Split_raw_string, textView, _undoHistoryRegistry, _editorOperationsFactoryService);
+
+        var edit = subjectBuffer.CreateEdit();
+
+        // Add a newline at the position of the end literal
+        var closingStart = GetStartPositionOfClosingDelimiter(token);
+        var newLineAndIndentation = newLine + indentation;
+        var insertedLines = 1;
+        edit.Insert(closingStart, newLineAndIndentation);
+        // Add a newline at the requested position
+        edit.Insert(position, newLineAndIndentation);
+        // Also add a newline at the start of the text, only if there is text before the requested position
```

Comment:
Why only then?

Reply:
Because the user might be trying to convert the single-line into a multi-line raw string, so they'd naturally position the cursor right after the starting delimiter to press enter. That's how I've converted mine too.

Reply:
But then, wouldn't they want anewline added in that case as well?

Reply:
The logic is to follow as shown in the tests:

In `"""$$text"""` if I press enter it's more likely I want to just make it a multi-line raw string than to enter a new-line before the text. If I want to add two newlines there instead of one, I'm still capable of pressing enter twice to add the second newline. The purpose of this command is to auto-format the closing delimiter to avoid the compiler error. 

If I put the cursor in between text and press enter, I will definitely need to reformat my raw string literal into a multi-line one so it's absolutely necessary to insert two extra newlines, one after the opening delimiter, and one before the closing one.

Reply:
I'm confused though. It says it only adds the newline at the start of there is text before.  But in both your examples, you want to add a newline at the start

Reply:
Oh I see the comment being confusing. With "before the requested position" I mean the caret position, I'll update it to be more clear.

Reply:
Ah. I see the confusion. Thanks

Reply:
Clarified comment